### PR TITLE
Fix for #496

### PR DESCRIPTION
**Pull request by tejksat**
I evaded from writing an integration test and just checked that no bytes from 0 to 255 are missed when using `HttpResponseStreamHandler`. Added Mockito as a test dependency for easy mocking `ChannelHandlerContext`.

---
This change is [Review on Reviewable](https://reviewable.io/reviews/docker-java/docker-java/501)

### CODE REVIEW

File: src/test/java/com/github/dockerjava/netty/handler/HttpResponseStreamHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    @Test
+    public void testNoBytesSkipped() throws Exception {
+        ResultCallbackTest callback = new ResultCallbackTest();
+        HttpResponseStreamHandler streamHandler = new HttpResponseStreamHandler(callback);
+        ChannelHandlerContext ctx = Mockito.mock(ChannelHandlerContext.class);
+        ByteBuf buffer = generateByteBuf();
+        streamHandler.channelRead0(ctx, buffer);
+        streamHandler.channelReadComplete(ctx);
+
+        Assert.assertTrue(IOUtils.contentEquals(callback.getInputStream(), new ByteBufInputStream(buffer)));
```
Comment (KostyaSha) [inline src/test/java/com/github/dockerjava/netty/handler/HttpResponseStreamHandlerTest.java]: please use static import

**Comment by KostyaSha** [comment]
@marcuslinke ?

**Comment by KostyaSha** [comment]
I think real integration test is also required.

**Comment by KostyaSha** [comment]
@tejksat please point "~/.docker-java.properties" (example could be found in resources) to some docker daemon and try implement integration test. Example could be found for any existing command.

**Comment by tejksat** [comment]
@KostyaSha ok, I'll check it a bit later.

**Comment by tejksat** [comment]
@KostyaSha done. Please, take a look.

**Comment by KostyaSha** [comment]
Cool, thanks. Would like @marcuslinke to look at.

**Comment by KostyaSha** [comment]
@marcuslinke ping

**Comment by marcuslinke** [comment]
Finally back in town hopefully I'm able to merge this evening.

**Comment by marcuslinke** [comment]
I will release 3.0.0-RC4 including this fix soon.

**Comment by marcuslinke** [comment]
@tejksat Thanks for contributing!

**Comment by tejksat** [comment]
Nice, thank you!

**Pull request merged**